docs/chymyst07.md

```diff
@@ -783,10 +793,173 @@ go { case barrier1(x1, reply1) + barrier2(x2, reply2) => ... }
+
+Note that the `counter(7)` molecule was emitted by a previous reaction of the same kind, 
+
+`barrier() + counter(6) => ... counter(7); ... }`
+
+So, this previous reaction is now blocked at the place where it emitted `counter(7)`. 
+
+Next, the reaction `barrier() + counter(7) => ...` will start and consume its input molecules, leaving two `barrier()` molecules present.
+
+Now the reaction body of `barrier() + counter(7) => ...` will run and emit `counter(8)`.
+Then we will have the molecules `barrier(), barrier(), counter(8)` in the soup.
+This set of molecules will be the same whether we define `reaction1` or `reaction2`.
+However, in `reaction1` the reply is already sent to `counter(7)` before emitting `counter(8)`.
+This will unblock the previous reaction,
+ 
+```scala
+{ case barrier(_, replyB) + counter(6, replyC) => replyC(); counter(7); replyB() }
+
+```
+
+which was blocked at the call to `counter(7)`.
+Now this reaction can proceed to reply to `barrier()` using `replyB()`.
+However, at this point it is too early to send a reply to `barrier()`, because we still have two `barrier()` molecules that we have not yet consumed!
+
+Therefore, `reaction1` is incorrect.
+On the other hand, `reaction2` works correctly.
+It will block at each emission of `counter(k+1)` and unblock only when `k=n-1`.
+At that time, `reaction2` will reply to the `counter(n-1)` molecule and to the last consumed `barrier()` molecule.
+The reply to the `counter(n-1)` molecule will unblock the copy of `reaction2` that emitted it, which will unblock the next molecules.
+
+In this way, the `n`-rendezvous will require all `n` reactions to wait at the "barrier" until all participants reach the barrier, and then unblock all of them.
+
+It remains to see how the first reaction will start.
+We could emit a `counter(0)` molecule at the initial time.
+This molecule is blocking, so emitting it will block a thread until the very end of the `n`-rendezvous.
+We could avoid blocking a thread by writing a special initial reaction such as
+
+```scala
+go { case barrier(_, replyB) + counterInit(_) => counter(1); replyB() }
+
+```
+
+The complete code looks like this:
+
+```scala
+val barrier = b[Unit,Unit]
+val counterInit = m[Unit]
+val counter = b[Int,Unit]
+
+site(
+  go { case barrier(_, replyB) + counterInit(_) => // this reaction will consume the very first barrier molecule emitted
+    counter(1) // one reaction has reached the rendezvous point
+    replyB()
+  },
+  go { case barrier(_, replyB) + counter(k, replyC) =>
+    if (k + 1 < n) counter(k+1)
+    replyC()
+    replyB()
+  }
+)
+counterInit() // This needs to be emitted initially.
+
+```
+
+Note that this chemistry will block `n` threads that emit `barrier()` and another set of `n` threads that run `reaction2`.
+Unless (in a future version of `JoinRun`) the blocking is optimized away, the current implementation of the `n`-rendezvous will require `2*n` threads and block them until the rendezvous is passed.
```

Comment:
your usage of parentheses after unless is unconventional. You could write


Unless the blocking is optimized away as may be the case in a future version of `JoinRun`, the ~~current~~ implementation of the ...